lib/trino-filesystem-alluxio/src/main/java/io/trino/filesystem/alluxio/AlluxioFileSystemCacheModule.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.alluxio;
+
+import alluxio.client.file.cache.CacheManager;
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.AlluxioProperties;
+import alluxio.conf.InstancedConfiguration;
+import alluxio.conf.PropertyKey;
+import alluxio.metrics.MetricsConfig;
+import alluxio.metrics.MetricsSystem;
+import com.google.inject.Binder;
+import com.google.inject.Inject;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.trino.filesystem.cache.TrinoFileSystemCache;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import static com.google.inject.Scopes.SINGLETON;
+import static com.google.inject.multibindings.MapBinder.newMapBinder;
+import static io.airlift.configuration.ConfigBinder.configBinder;
+
+public class AlluxioFileSystemCacheModule
+        extends AbstractConfigurationAwareModule
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        configBinder(binder).bindConfig(AlluxioFileSystemCacheConfig.class);
+
+        newMapBinder(binder, String.class, TrinoFileSystemCache.class).addBinding("alluxio").to(AlluxioFileSystemCache.class).in(SINGLETON);
+
+        Properties metricProps = new Properties();
+        metricProps.put("sink.jmx.class", "alluxio.metrics.sink.JmxSink");
+        metricProps.put("sink.jmx.domain", "org.alluxio");
+        MetricsSystem.startSinksFromConfig(new MetricsConfig(metricProps));
+    }
+
+    @Inject
+    @Provides
+    @Singleton
+    public static AlluxioConfiguration getAlluxioConfiguration(AlluxioFileSystemCacheConfig config)
+    {
+        AlluxioProperties alluxioProperties = new AlluxioProperties();
+        alluxioProperties.set(PropertyKey.USER_CLIENT_CACHE_ENABLED, true);
+        alluxioProperties.set(PropertyKey.USER_CLIENT_CACHE_DIRS, config.getCacheDirectories());
+        alluxioProperties.set(PropertyKey.USER_CLIENT_CACHE_SIZE, config.getMaxCacheSize());
+        alluxioProperties.set(PropertyKey.USER_CLIENT_CACHE_SHADOW_ENABLED, true);
```

Comment:
cc @beinan : this config was set in the original Alluxio PR so I assume it is needed, but I'm unsure why, could you enlighten us?

Fwiw, I don't understand [the Alluxio definition](https://docs.alluxio.io/os/user/stable/en/reference/Properties-List.html) either 😅 
> If this is enabled, a shadow cache will be created to tracking the working set of a past time window, and measure the hit ratio if the working set fits the cache

Reply:
It looks like it's a structure for tracking working set size and cache hit rate, if the cache had infinite space. Please see
https://www.alluxio.io/blog/improving-presto-architectural-decisions-with-alluxio-shadow-cache-at-meta-facebook/
It will be useful for tuning cache sizes. We could make it optional?